storage_controller/src/http.rs

```diff
@@ -1005,6 +1006,29 @@ async fn handle_tenant_shard_migrate(
     )
 }
 
+async fn handle_tenant_shard_migrate_secondary(
+    service: Arc<Service>,
+    req: Request<Body>,
+) -> Result<Response<Body>, ApiError> {
+    check_permissions(&req, Scope::Admin)?;
+
+    let mut req = match maybe_forward(req).await {
+        ForwardOutcome::Forwarded(res) => {
+            return res;
+        }
+        ForwardOutcome::NotForwarded(req) => req,
+    };
+
+    let tenant_shard_id: TenantShardId = parse_request_param(&req, "tenant_shard_id")?;
+    let migrate_req = json_request::<TenantShardMigrateRequest>(&mut req).await?;
```

Comment:
It seems problematic that we're taking the tenant shard ID both from a request parameter and also from the request body -- which one is it? It doesn't seem like `TenantShardMigrateRequest.tenant_shard_id` is actually read anywhere, so let's get rid of it (possibly in a separate PR).

Reply:
Yeah, this is pure tech debt and was never read.  Removed in 1af991c2d

Reply:
Let's just be careful with backwards-compatibility here, e.g. if the deserialization errors on unknown fields.

Reply:
yep, should be fine -- serde json allows unknown fields by default